src/openssh.c

```diff
@@ -108,85 +127,98 @@ static bool openssh_write_secret(const char *output_directory, const char *usern
-    buffer_writer_write_uint32(body_writer, 1);
+    ok &= buffer_writer_write_uint32(body_writer, 1);
 
     // Public Keys Length.
-    buffer_writer_write_uint32(body_writer, 51);
+    ok &= buffer_writer_write_uint32(body_writer, 51);
 
     // Public Key.
-    buffer_writer_write_uint32(body_writer, 11);
-    buffer_writer_write_value(body_writer, "ssh-ed25519", 11);
-    buffer_writer_write_uint32(body_writer, 32);
-    buffer_writer_write_value(body_writer, public, 32);
+    ok &= buffer_writer_write_uint32(body_writer, 11);
+    ok &= buffer_writer_write_value(body_writer, "ssh-ed25519", 11);
+    ok &= buffer_writer_write_uint32(body_writer, 32);
+    ok &= buffer_writer_write_value(body_writer, public, 32);
 
     // Private key section
-    uint32_t checkint = randombytes_random();
 
-    buffer_writer_write_uint32(body_writer, private_length + padding);
+    // {checkint} is a random integer used to identify a correctly decrypted
+    // private key in cases where the private key is encrypted. ours are not.
+    const uint32_t checkint = 0x12345678;
+
+    ok &= buffer_writer_write_uint32(body_writer, private_length + padding);
 
-    buffer_writer_write_uint32(body_writer, checkint);
-    buffer_writer_write_uint32(body_writer, checkint);
+    ok &= buffer_writer_write_uint32(body_writer, checkint);
+    ok &= buffer_writer_write_uint32(body_writer, checkint);
 
-    buffer_writer_write_uint32(body_writer, 11);
-    buffer_writer_write_value(body_writer, "ssh-ed25519", 11);
-    buffer_writer_write_uint32(body_writer, 32);
-    buffer_writer_write_value(body_writer, public, 32);
+    ok &= buffer_writer_write_uint32(body_writer, 11);
+    ok &= buffer_writer_write_value(body_writer, "ssh-ed25519", 11);
+    ok &= buffer_writer_write_uint32(body_writer, 32);
+    ok &= buffer_writer_write_value(body_writer, public, 32);
 
-    buffer_writer_write_uint32(body_writer, 64);
-    buffer_writer_write_value(body_writer, secret, 64);
+    ok &= buffer_writer_write_uint32(body_writer, 64);
+    ok &= buffer_writer_write_value(body_writer, secret, 64);
 
-    buffer_writer_write_uint32(body_writer, username_length);
-    buffer_writer_write_value(body_writer, username, username_length);
+    ok &= buffer_writer_write_uint32(body_writer, username_length);
+    ok &= buffer_writer_write_value(body_writer, username, username_length);
 
     uint8_t pad = 0;
     while (padding--)
-        buffer_writer_write_uint8(body_writer, ++pad & 0xff);
+        ok &= buffer_writer_write_uint8(body_writer, ++pad & 0xff);
+
+    buffer_writer_free(body_writer);
 
     // BASE64 encode the buffer.
-    buffer_base64_encode(body, &body_base64);
+    ok &= buffer_base64_encode(body, &body_base64);
 
-    write(fd, "-----BEGIN OPENSSH PRIVATE KEY-----\n", 36);
+    struct buffer * fd_buf = buffer_new(
+      36   // "BEGIN OPENSSH PRIVATE KEY..."
+      + body_base64->size
+      + (body_base64->size / 70) + (body_base64->size % 70) // newlines
```

Comment:
Oops, this should have been rounded up division rather than the modulo I put in there, sorry!
```
+ (body_base64->size + (70 -1) ) / 70
```